services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -64,25 +64,79 @@ class SwitchRulesSpec extends BaseSpecification {
-        // this scenario. It is needed to add additional functionality to properly disconnect the switch.
+    def "Pre-installed rules (including default) are not deleted from a new switch connected to the controller"() {
+        requireProfiles("virtual")
+
+        given: "A switch with some rules installed (including default) and not connected to the controller"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.getActiveSwitches()[0..1]
+        def defaultRules = northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.sort { it.cookie }
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        def defaultPlusFlowRules = []
+        assert Wrappers.wait(WAIT_OFFSET) {
+            defaultPlusFlowRules = northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.sort { it.cookie }
+            defaultPlusFlowRules.size() == defaultRules.size() + FLOW_RULES_SIZE
+        }
+
+        lockKeeperService.knockoutSwitch(srcSwitch.dpId)
+        assert Wrappers.wait(discoveryTimeout + rerouteDelay + WAIT_OFFSET * 2) {
+            northboundService.getFlowStatus(flow.id).status == FlowState.DOWN
+        }
+
+        northboundService.deleteFlow(flow.id)
+        assert Wrappers.wait(WAIT_OFFSET) { !(flow.id in northboundService.getAllFlows()*.id) }
 
+        when: "Connect the switch to the controller"
+        lockKeeperService.reviveSwitch(srcSwitch.dpId)
+        assert Wrappers.wait(WAIT_OFFSET) { srcSwitch.dpId in northboundService.getActiveSwitches()*.switchId }
+
+        then: "Previously installed rules are not deleted from the switch"
+        def actualRules = northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.sort { it.cookie }
+        actualRules.size() == defaultPlusFlowRules.size()
+        [actualRules, defaultPlusFlowRules].transpose().each { actual, expected ->
+            verifyAll {
+                actual.cookie == expected.cookie
+                actual.tableId == expected.tableId
+                actual.version == expected.version
+                actual.priority == expected.priority
+                actual.idleTimeout == expected.idleTimeout
+                actual.hardTimeout == expected.hardTimeout
+                actual.match == expected.match
+                actual.instructions == expected.instructions
+                actual.flags == expected.flags
+            }
+        }
+
+        and: "Delete previously installed rules"
+        northboundService.deleteSwitchRules(srcSwitch.dpId, DeleteRulesAction.IGNORE_DEFAULTS)
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.size() == defaultRules.size()
+        }
+    }
+
+    def "Pre-installed rules are not deleted from a new switch connected to the controller"() {
```

Comment:
How is this different from the previous test?

Reply:
Yes, I also thought whether we need this test or not. Would you like me to delete this test? Or we can just merge them as I did in my next PRs for other tests?

Reply:
I didn't get the actual difference at this point. Is it the same but without flow rules? If yes, then I think it is fully covered by the above one